huaweicloud/services/cce/resource_huaweicloud_cce_chart.go

```diff
@@ -0,0 +1,216 @@
+package cce
+
+import (
+	"context"
+	"os"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/huaweicloud/huaweicloud-sdk-go-v3/core/def"
+	cce "github.com/huaweicloud/huaweicloud-sdk-go-v3/services/cce/v3/model"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+// API: CCE POST /v2/charts
+// API: CCE GET /v2/charts/{chart_id}
+// API: CCE PUT /v2/charts/{chart_id}
+// API: CCE DELETE /v2/charts/{chart_id}
+
+func ResourceChart() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceChartCreate,
+		ReadContext:   resourceChartRead,
+		UpdateContext: resourceChartUpdate,
+		DeleteContext: resourceChartDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"content": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"parameters": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"value": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"translate": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"instruction": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"source": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"public": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"chart_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceChartCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	client, err := cfg.HcCceV3Client(cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating CCE v3 client: %s", err)
+	}
+
+	file, err := os.Open(d.Get("content").(string))
+	if err != nil {
+		diag.Errorf("error opening chart file: %s", err)
+	}
+	defer file.Close()
+
+	createOpts := cce.UploadChartRequestBody{
+		Parameters: &def.MultiPart{
+			Content: d.Get("parameters").(string),
+		},
+		Content: &def.FilePart{
+			Content: file,
+		},
+	}
+
+	req := cce.UploadChartRequest{
+		Body: &createOpts,
+	}
+	resp, err := client.UploadChart(&req)
+	if err != nil {
```

Comment:
should we check the `resp` as it is a pointer?

Reply:
sure